### PR TITLE
[UX] Gesture manager: add action - toggle wifi

### PR DESCRIPTION
**Pull request by Frenzie**
As suggested by @poire-z.

https://github.com/koreader/koreader/issues/4727#issuecomment-469668338

I wanted to make the gesture a `W`, but that's a bit too unwieldy. Instead there's the suggestion of a (backward) `W`.

g1048

### CODE REVIEW

File: frontend/apps/reader/modules/readergesture.lua [hunk 1]
```diff
@@ -697,6 +701,20 @@ function ReaderGesture:gestureAction(action, ges)
...
+
+        if not NetworkMgr:isOnline() then
+            -- NB Normal widgets should use NetworkMgr:promptWifiOn()
+            -- This is specifically the toggle wifi action, so consent is implied.
+            NetworkMgr:turnOnWifi()
+        else
+            NetworkMgr:turnOffWifi()
+
+            UIManager:show(InfoMessage:new{
+                text = _("Wifi disabled."),
```
Comment (poire-z) [inline frontend/apps/reader/modules/readergesture.lua]: No timeout to this InfoMessage? If we use gestures, it's that we want things fast and get done :) No need for this additional tap to dismiss the InfoMessage I think.
I would even use a Notification with timeout, for even less disruption.

Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: Right, good point.

I was testing this and had no idea if turning it off worked or not without going to the menu, which seemed to defeat the point.

**Comment by Frenzie** [comment]
As a default gesture or just as available actions?

**Comment by poire-z** [comment]
As alternate available actions. Some users may prefer having two different gestures, some other only one that toggles I think.
No opinion about defaults or which of these should be set as default.

File: frontend/apps/reader/modules/readergesture.lua [hunk 2]
```diff
@@ -330,6 +339,9 @@ function ReaderGesture:buildMenu(ges, default)
         {"toggle_gsensor", Device:canToggleGSensor()},
         {"toggle_rotation", not self.is_docless, true},
         {"toggle_reflow", not self.is_docless, true},
+        {"toggle_wifi", Device:hasWifiToggle() and "toggle_wifi", true},
+        {"wifi_on", Device:hasWifiToggle() and "wifi_on", true},
+        {"wifi_off", Device:hasWifiToggle() and "wifi_off", true},
```
Comment (poire-z) [inline frontend/apps/reader/modules/readergesture.lua]: Looks strange. Shouldn't they be `{string action_name, boolean enabled, boolean separator}` ?
And I guess the separator could be a bit moved with the added stuff.

File: frontend/apps/reader/modules/readergesture.lua [hunk 3]
```diff
@@ -339,9 +339,9 @@ function ReaderGesture:buildMenu(ges, default)
         {"toggle_gsensor", Device:canToggleGSensor()},
         {"toggle_rotation", not self.is_docless, true},
         {"toggle_reflow", not self.is_docless, true},
-        {"toggle_wifi", Device:hasWifiToggle() and "toggle_wifi", true},
-        {"wifi_on", Device:hasWifiToggle() and "wifi_on", true},
-        {"wifi_off", Device:hasWifiToggle() and "wifi_off", true},
+        {"toggle_wifi", Device:hasWifiToggle(), true},
+        {"wifi_on", Device:hasWifiToggle(), true},
+        {"wifi_off", Device:hasWifiToggle(), true},
```
Comment (poire-z) [inline frontend/apps/reader/modules/readergesture.lua]: Still too many separators :)
All the 3 wifi ones should be in the same separated area.
And I guess you want to move toggle_reflow below them, near the Zoom to fit as it's kopt stuff

Comment (Frenzie) [reply frontend/apps/reader/modules/readergesture.lua]: Depends on whether you see it as the "toggle section" or the "zoom-type stuff section". I suppose "zoom stuff" makes more sense as a grouping.

**Comment by Frenzie** [comment]
(No change, just had to manually merge it with the rotation toggle change.)

File: frontend/apps/reader/modules/readergesture.lua [hunk 4]
```diff
@@ -118,6 +123,9 @@ local default_multiswipes = {
     "northeast southeast",
     -- "southwest northwest", -- visually ambiguous
     -- "northwest southwest", -- visually ambiguous
+    "northwest southwest northwest",
```
Comment (poire-z) [inline frontend/apps/reader/modules/readergesture.lua]: That one is as ambiguous as the previous one I didn't dare added :) So, we might as well dare.

Comment (poire-z) [reply frontend/apps/reader/modules/readergesture.lua]: Mhhh, actually it is not :) it reads the same backwards following the arrows :)

**Pull request merged**